javascript/index.js

```diff
@@ -4,37 +4,143 @@
-  
-  getInstruction("mashedPotatoes", 1, (step2) => {
+  },
+  (error) => console.log(error)
+);
+
+getInstruction(
+  "mashedPotatoes",
+  1,
+  (step2) => {
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 2, (step3) => {
+  },
+  (error) => console.log(error)
+);
+
+getInstruction(
+  "mashedPotatoes",
+  2,
+  (step3) => {
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 3, (step4) => {
+  },
+  (error) => console.log(error)
+);
+
+getInstruction(
+  "mashedPotatoes",
+  3,
+  (step4) => {
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 4, (step5) => {
+  },
+  (error) => console.log(error)
+);
+
+getInstruction(
+  "mashedPotatoes",
+  4,
+  (step5) => {
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
     document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
-  }, (error) => console.log(error));
-
-
+  },
+  (error) => console.log(error)
+);
 
 // Iteration 1 - using callbacks
 // ...
 
-// Iteration 2 - using promises
-// ...
+// Iteration 2 - using promises + image (Bonus 1)
 
-// Iteration 3 using async/await
-// ...
+obtainInstruction("steak", 0)
+  .then((step1) => {
+    document.querySelector("#steak").innerHTML += `<li>${step1}</li>`;
+    return obtainInstruction("steak", 1);
+  })
+  .then((step2) => {
+    document.querySelector("#steak").innerHTML += `<li>${step2}</li>`;
+    return obtainInstruction("steak", 2);
+  })
+  .then((step3) => {
+    document.querySelector("#steak").innerHTML += `<li>${step3}</li>`;
+    return obtainInstruction("steak", 3);
+  })
+  .then((step4) => {
+    document.querySelector("#steak").innerHTML += `<li>${step4}</li>`;
+    return obtainInstruction("steak", 4);
+  })
+  .then((step5) => {
+    document.querySelector("#steak").innerHTML += `<li>${step5}</li>`;
+    return obtainInstruction("steak", 5);
+  })
+  .then((step6) => {
+    document.querySelector("#steak").innerHTML += `<li>${step6}</li>`;
+    return obtainInstruction("steak", 6);
+  })
+  .then((step7) => {
+    document.querySelector("#steak").innerHTML += `<li>${step7}</li>`;
+    return obtainInstruction("steak", 7);
+  })
+  .then((ready) => {
+    ready = document.querySelector(
+      "#steak"
+    ).innerHTML += `<li>Steak is ready!</li>`;
+    const steakImage = document.querySelector("#steakImg");
+    steakImage.removeAttribute("hidden");
+    return obtainInstruction("steak", 8);
+  });
+
+// Iteration 3 using async/await + image (Bonus 1)
+
+async function makeBroccoli() {
+  for (let i = 0; i <= broccoli.length - 1; i++) {
+    document.querySelector("#broccoli").innerHTML += `<li>${broccoli[i]}</li>`;
+    try {
```

Comment:
nice try catch logic. you know, you could use the error in the catch to stop if there are no more instructions to be obtained, then you wouldn't even need to loop until broccoli.length but could simply while loop, increasing i++ until the error breaks the loop :)